app/form_objects/result_filters/location_filter_form.rb

```diff
@@ -0,0 +1,82 @@
+module ResultFilters
+  class LocationFilterForm
+    NO_OPTION = nil
+    LOCATION_OPTION = "1".freeze
+
+    attr_reader :params, :errors
+
+    def initialize(params)
+      @params = params
+      @errors = nil
+      @valid = nil
+      validate
+    end
+
+    def valid?
+      @valid
```

Comment:
I think we'd usually call `validate` in here and then return `errors.empty?`. It would be good to keep this as `ActiveModel` as possible so as it has a familiar interface. You can include `ActiveModel::Validations` and then do things like `validates :option, :presence` etc. You can also use `ActiveModel::Errors` although that might not help you much here. [(example)](https://api.rubyonrails.org/classes/ActiveModel/Errors.html)

Reply:
@gpeng - I've made the change but not sure about including `ActiveModel::Validations` at this stage - think it's potentially overkill?